cpp/modmesh/buffer/SimpleArray.hpp

```diff
@@ -312,46 +312,31 @@ class SimpleArray
     explicit SimpleArray(small_vector<size_t> const & shape,
                          small_vector<size_t> const & stride,
                          std::shared_ptr<buffer_type> const & buffer,
-                         bool is_c_contiguous = true,
-                         bool is_f_contiguous = false)
+                         bool c_contiguous,
+                         bool f_contiguous)
+        : SimpleArray(shape, stride, buffer)
```

Comment:
In order to avoid `error: call to constructor of 'SimpleArray<bool>' is ambiguous
  330 |         return SimpleArray(shape, stride, buffer);`, I don't assign the default value to `c_contiguous` and `f_contiguous`.

Reply:
Since we only have C type or F type, why not just:

```c++
    explicit SimpleArray(small_vector<size_t> const & shape,
                         small_vector<size_t> const & stride,
                         std::shared_ptr<buffer_type> const & buffer,
                         bool is_c_contiguous = true)
```

I prefer using enum more, just for your reference:

```c++
enum class SimpleArrayMemoryType {
    CType
    FType
};

    explicit SimpleArray(small_vector<size_t> const & shape,
                         small_vector<size_t> const & stride,
                         std::shared_ptr<buffer_type> const & buffer,
                         SimpleArrayMemoryType is_c_contiguous = SimpleArrayMemoryType::CType)
```

Reply:
Boolean flag does not need to introduce new symbols at the Python side and is preferred ATM.

The input buffer may be F or C contiguous or discontinues.  A single Boolean flag is not enough.

Reply:
And it's good to remove default arguments in C++.  They usually do harm more than good.

Reply:
On the Python side, the argument can be a string; we already have the example from `DataType`.

Reply:
By the way, if the array is 1 dimension, it could be c-contiguous and f-contiguous simultaneously, and I'm not sure enum is a good manner.